bin/carthage-bootstrap-if-needed

```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env sh
+
+if ! diff Cartfile.resolved Carthage/Cartfile.resolved &>/dev/null; then
+  carthage bootstrap 
+  cp Cartfile.resolved Carthage
+fi
```

Comment:
We're copying the `Cartfile.resolved` into the `Carthage` directory on line 5. That directory is then cached by Circle (specified in circle.ml)

Although, it occurs to me that since we're using git submodules, we don't actually need to do _any_of this, so I'm going to remove it.